.eslint/rules/icon-has-viewBox.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * @fileoverview All Paste icon svgs should have a viewBox attribute
+ * @author Glorilí Alejandro
+ */
+
+const getJSXIdentifier = (node) => {
+  if (node === undefined) {
+    return;
+  }
+  if (node.type === 'JSXIdentifier') {
+    return node;
+  }
+  if (node.name !== undefined) {
+    return getJSXIdentifier(node.name);
+  }
+  if (node.openingElement !== undefined) {
+    return getJSXIdentifier(node.openingElement);
+  }
+  return undefined;
+};
+
+module.exports = {
+  meta: {
+    type: 'problem',
+  },
+
+  create(context) {
+    return {
+      // this lets us work with all the JSXElement nodes
+      JSXElement(node) {
+        // I want to get the openingElement node of each, since that's
+        // where the details about both it being an svg and the
+        // attributes live.
+        const elementDetails = node.openingElement;
+
+        // To get find the name, we need to go into the openingElement
+        // we just got, get its JSXIdentifier, and then get that node's name
+        if ((getJSXIdentifier(elementDetails) || {}).name === 'svg') {
+          // there can be multiple attributes, so we go thru each
+          // to see if any are viewBox
+          for (attribute of elementDetails.attributes) {
+            // we again need to go into each attribute node and get its
+            // JSXIdentifier and then its name.
+            if (getJSXIdentifier(attribute).name === 'viewBox') {
+              // if any of them are 'viewBox', we are done here.
+              return;
+            }
+          }
+          // if we are in an svg node (which in here, we know we are) and
+          // we haven't exited out already from viewBox being present,
+          // then we need to report this node. Bad node!
+          context.report({
+            node,
+            message: 'Every icon needs to have a viewbox attribute',
```

Comment:
thought: looking at the code, this is checking that any `svg` has a `viewBox` attribute. SVGs can be used for more than icons, such as for illustrations. It may be confusing that the wording here is "icon" if this reports for a non-icon SVG.

Reply:
This is a great point, I definitely want to make this message better. I'm thinking of changing it to "An svg must have a viewBox". Thoughts?

Reply:
I like it!